### PR TITLE
[FE] Added Profile page 

### PR DESCRIPTION
**Pull request by ZhimingZhao**
## Changes:

- Added Profile Page module
- Linked start page and profile page

image

## TODO:
- Add player name and profile picture from auth service (add a function in auth service)
- Pull player data from users (needs to be worked on by me)
- Pull match data from match (needs to be worked on by me)

### CODE REVIEW

File: src/frontend/src/app/myprofile/myprofile.component.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-myprofile',
+  templateUrl: './myprofile.component.html',
+  styleUrls: ['./myprofile.component.scss'],
+})
+export class MyprofileComponent {
+  id: string = '';
+  players: PlayerData[] = [];
```
Comment (YoussefRizkalla) [inline src/frontend/src/app/myprofile/myprofile.component.ts:12]: There'll only be one player here, so no need to use a list.

File: src/frontend/src/app/myprofile/myprofile.component.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import { PlayerData } from '../types/PlayerData';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-myprofile',
+  templateUrl: './myprofile.component.html',
+  styleUrls: ['./myprofile.component.scss'],
+})
+export class MyprofileComponent {
+  id: string = '';
```
Comment (YoussefRizkalla) [inline src/frontend/src/app/myprofile/myprofile.component.ts:11]: Do we need this?

File: src/frontend/src/app/myprofile/myprofile.component.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { Component, OnInit } from '@angular/core';
+import { PlayerData } from '../types/PlayerData';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-myprofile',
+  templateUrl: './myprofile.component.html',
+  styleUrls: ['./myprofile.component.scss'],
+})
+export class MyprofileComponent {
```
Comment (YoussefRizkalla) [inline src/frontend/src/app/myprofile/myprofile.component.ts:10]: nit: MyProfileComponent

File: src/frontend/src/app/myprofile/myprofile.component.scss [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  column-gap: 2%;
+}
+
+.left-half {
+  background: rgba(255, 255, 255, 0.05);
+  grid-column: 1;
+}
+
+.right-half {
+  //background: rgba(255, 255, 255, 0.05);
```
Comment (YoussefRizkalla) [inline src/frontend/src/app/myprofile/myprofile.component.scss:49]: Remove if not needed

File: src/frontend/src/app/myprofile/myprofile.component.html [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+</div>
+<div align="center">
+  <h1 class="title">Player1</h1>
+</div>
+<section class="container">
+  <p class="stats-title">Stats</p>
+  <p class="stats-title">Match History</p>
+  <div class="left-half">
+    <mat-card class="stats-card">
+      <h1 class="stat-text1">Wins: 10</h1>
```
Comment (YoussefRizkalla) [inline src/frontend/src/app/myprofile/myprofile.component.html:13]: Not needed for now but would make it easier for the feature - add those variables to a type and reference them here (hardcode the actual values in the .ts for now, or hardcode them in a service which will be replaced with a network call later).

Comment (ZhimingZhao) [reply src/frontend/src/app/myprofile/myprofile.component.html]: image
image

File: src/frontend/src/app/myprofile/myprofile.component.ts (hunk 1 above)
Comment (ZhimingZhao) [reply src/frontend/src/app/myprofile/myprofile.component.ts]: Removed list

File: src/frontend/src/app/myprofile/myprofile.component.ts (hunk 2 above)
Comment (ZhimingZhao) [reply src/frontend/src/app/myprofile/myprofile.component.ts]: Removed and added type instead

File: src/frontend/src/app/myprofile/myprofile.component.ts (hunk 3 above)
Comment (ZhimingZhao) [reply src/frontend/src/app/myprofile/myprofile.component.ts]: Changed to MyProfileComponent

File: src/frontend/src/app/myprofile/myprofile.component.scss (hunk 4 above)
Comment (ZhimingZhao) [reply src/frontend/src/app/myprofile/myprofile.component.scss]: Removed unused code

File: src/frontend/src/app/types/PlayerData.ts [hunk 6]
```diff
@@ -3,4 +3,10 @@ export type PlayerData = {
   name: string;
   picture: string;
   finishedCurrentRound: boolean;
+  Wins: number;
```
Comment (YoussefRizkalla) [inline src/frontend/src/app/types/PlayerData.ts:6]: Let's use lower camel case for these too (i.e. wins, problemSolved, etc)

**Review by YoussefRizkalla — APPROVED** [review]

**Comment by YoussefRizkalla** [comment]
Closes #29

**Review by AntonKanug — APPROVED** [review]

**Pull request merged**